### PR TITLE
Add Except support for Materialized view

### PR DESCRIPTION
**Pull request by pranjalssh**
Test plan - Added test

https://github.com/prestodb/presto/issues/16608

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java [hunk 1]
```diff
@@ -1562,6 +1562,38 @@ public void testMaterializedViewForUnionAll()
...
+                    "AS %s", view, baseQuery));
+
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE nationkey < 10", view, baseQuery), 380);
+
+            String viewQuery = format("SELECT name, custkey, nationkey from %s ORDER BY name", view);
+            baseQuery = format("%s ORDER BY name", baseQuery);
+
+            MaterializedResult viewTable = computeActual(viewQuery);
+            MaterializedResult baseTable = computeActual(baseQuery);
+            assertEquals(viewTable, baseTable);
```
Comment (jainxrohit) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1589]: Lets add the plan comparison as well.

**Review by jainxrohit — CHANGES_REQUESTED** [review]
> Test plan - Added test
> 
> ```
> == RELEASE NOTES ==
> 
> General Changes
> * Materialized views now work with EXCEPT in query clause
> ```

Materialized view has been released yet. So lets keep as no release notes.


change the commit message as 
Add Except support for Materialized view

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java [hunk 2]
```diff
@@ -1562,6 +1562,38 @@ public void testMaterializedViewForUnionAll()
...
+                    "AS SELECT custkey, name, address, nationkey FROM customer", table));
+
+            String baseQuery = format(
+                    "SELECT name, custkey, nationkey FROM %s WHERE custkey < 1000 EXCEPT " +
+                            "SELECT name, custkey, nationkey FROM %s WHERE custkey > 900", table, table);
+
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['nationkey']) " +
+                    "AS %s", view, baseQuery));
+
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE nationkey < 10", view, baseQuery), 380);
```
Comment (zacw7) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1582]: Here only one argument is expected for the format string but found two.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java [hunk 3]
```diff
@@ -1562,6 +1562,38 @@ public void testMaterializedViewForUnionAll()
...
+
+            String viewQuery = format("SELECT name, custkey, nationkey from %s ORDER BY name", view);
+            baseQuery = format("%s ORDER BY name", baseQuery);
```
Comment (zacw7) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1585]: Why is this `ORDER BY` needed?

**Review by zacw7 — CHANGES_REQUESTED** [review]
Looks good in general. Only a couple of nits.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java (hunk 3 above)
Comment (pranjalssh) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1585]: So we can assert results of both queries are same

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java (hunk 1 above)
Comment (pranjalssh) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1589]: Not sure if there's an easy way, the plan looks pretty complicated with exchanges happening for except.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java [hunk 4]
```diff
@@ -1562,6 +1562,38 @@ public void testMaterializedViewForUnionAll()
         }
     }
 
+    @Test
+    public void testMaterializedViewForExcept()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        String table = "test_customer_union";
```
Comment (gggrace14) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1569]: Do you want to name it test_customer_except?

**Review by gggrace14 — COMMENTED** [review]
The logics looks good to me. Please address @zacw7 and @jainxrohit 's comment, and we're good to go.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java (hunk 2 above)
Comment (gggrace14) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java:1582]: Forgot to change this?

**Comment by gggrace14** [comment]
Ready to merge

**Comment by pranjalssh** [comment]
Test failure looks unrelated

**Comment by highker** [comment]
@pranjalssh, ya, I will restart the tests. Unfortunately, we disabled merging commits with failed tests lol.

**Comment by pranjalssh** [comment]
@highker all tests pass now

**Pull request merged**